### ISSUE TITLE
The downloading of db for trivy may forbidden by Github.

### ISSUE THREAD

**Issue by heww** [goharbor/harbor#11031]
Install trivy as the default scanner for vulnerabilities, the trivy will download db from the Github. This may is forbidden by the Github when API rate limit exceeded. 

Maybe we need to provide configurations to set Github token for the trivy scanner to download the db.

```
Mar 11 13:12:45 192.168.80.1 trivy-adapter[21287]: {"exit_code":1,"image_ref":"core:8080/library/redis@sha256:16ece50399ba1eb8ed67e03687423434e5fa75c52a448f94519664e0a9e7829c","level":"error","msg":"Running trivy failed","std_err":"","std_out":"2020-03-11T13:12:45.594Z\t\u001b[34mINFO\u001b[0m\tNeed to update DB\n2020-03-11T13:12:45.594Z\t\u001b[34mINFO\u001b[0m\tDownloading DB...\n2020-03-11T13:12:45.964Z\t\u001b[31mFATAL\u001b[0m\tfailed to download vulnerability DB: failed to download vulnerability DB: failed to list releases: GET https://api.github.com/repos/aquasecurity/trivy-db/releases: 403 API rate limit exceeded for 66.170.99.2. (But here's the good news: Authenticated requests get a higher rate limit. Check out the documentation for more details.) [rate reset in 11m38s]\n","time":"2020-03-11T13:12:45Z"}
Mar 11 13:12:45 192.168.80.1 trivy-adapter[21287]: {"error":"running trivy wrapper: running trivy: exit status 1: ","level":"error","msg":"Scan failed","time":"2020-03-11T13:12:45Z"}
Mar 11 13:12:46 192.168.80.1 trivy-adapter[21287]: {"error":"running trivy wrapper: running trivy: exit status 1: ","level":"error","msg":"Scan job failed","scan_job_id":"15646a0566fefce4f064edaf","time":"2020-03-11T13:12:46Z"}
Mar 11 13:12:47 192.168.80.1 trivy-adapter[21287]: {"exit_code":1,"image_ref":"core:8080/library/redis@sha256:16ece50399ba1eb8ed67e03687423434e5fa75c52a448f94519664e0a9e7829c","level":"error","msg":"Running trivy failed","std_err":"","std_out":"2020-03-11T13:12:47.106Z\t\u001b[34mINFO\u001b[0m\tNeed to update DB\n2020-03-11T13:12:47.106Z\t\u001b[34mINFO\u001b[0m\tDownloading DB...\n2020-03-11T13:12:47.445Z\t\u001b[31mFATAL\u001b[0m\tfailed to download vulnerability DB: failed to download vulnerability DB: failed to list releases: GET https://api.github.com/repos/aquasecurity/trivy-db/releases: 403 API rate limit exceeded for 66.170.99.2. (But here's the good news: Authenticated requests get a higher rate limit. Check out the documentation for more details.) [rate reset in 11m38s]\n","time":"2020-03-11T13:12:47Z"}
Mar 11 13:12:47 192.168.80.1 trivy-adapter[21287]: {"error":"running trivy wrapper: running trivy: exit status 1: ","level":"error","msg":"Scan failed","time":"2020-03-11T13:12:47Z"}
Mar 11 13:12:48 192.168.80.1 trivy-adapter[21287]: {"error":"running trivy wrapper: running trivy: exit status 1: ","level":"error","msg":"Scan job failed","scan_job_id":"68aff9f4aa6bde2c595e5537","time":"2020-03-11T13:12:48Z"}
Mar 11 13:12:48 192.168.80.1 trivy-adapter[21287]: {"error":"running trivy wrapper: running trivy: exit status 1: ","level":"error","msg":"Scan job failed","scan_job_id":"662aab9019f16dced6020689","time":"2020-03-11T13:12:48Z"}
Mar 11 13:12:53 192.168.80.1 trivy-adapter[21287]: {"exit_code":1,"image_ref":"core:8080/library/redis@sha256:ee1d00c5250b5a886b09be2d5f9506add35dfb557f1ef37a7e4b8f0138f32956","level":"error","msg":"Running trivy failed","std_err":"","std_out":"2020-03-11T13:12:53.589Z\t\u001b[34mINFO\u001b[0m\tNeed to update DB\n2020-03-11T13:12:53.590Z\t\u001b[34mINFO\u001b[0m\tDownloading DB...\n2020-03-11T13:12:53.953Z\t\u001b[31mFATAL\u001b[0m\tfailed to download vulnerability DB: failed to download vulnerability DB: failed to list releases: GET https://api.github.com/repos/aquasecurity/trivy-db/releases: 403 API rate limit exceeded for 66.170.99.2. (But here's the good news: Authenticated requests get a higher rate limit. Check out the documentation for more details.) [rate reset in 11m30s]\n","time":"2020-03-11T13:12:53Z"}
Mar 11 13:12:53 192.168.80.1 trivy-adapter[21287]: {"error":"running trivy wrapper: running trivy: exit status 1: ","level":"error","msg":"Scan failed","time":"2020-03-11T13:12:53Z"}
Mar 11 13:12:56 192.168.80.1 trivy-adapter[21287]: {"error":"running trivy wrapper: running trivy: exit status 1: ","level":"error","msg":"Scan job failed","scan_job_id":"83d42c60bb5cfe0f8b080f76","time":"2020-03-11T13:12:56Z"}
Mar 11 13:13:05 192.168.80.1 trivy-adapter[21287]: {"exit_code":1,"image_ref":"core:8080/library/hello-world@sha256:271f28bd488f30b1651affeb9bdfe1bd9122c920ac4f7e9cc11631c3950426cd","level":"error","msg":"Running trivy failed","std_err":"","std_out":"2020-03-11T13:13:05.101Z\t\u001b[34mINFO\u001b[0m\tNeed to update DB\n2020-03-11T13:13:05.101Z\t\u001b[34mINFO\u001b[0m\tDownloading DB...\n2020-03-11T13:13:05.570Z\t\u001b[31mFATAL\u001b[0m\tfailed to download vulnerability DB: failed to download vulnerability DB: failed to list releases: GET https://api.github.com/repos/aquasecurity/trivy-db/releases: 403 API rate limit exceeded for 66.170.99.2. (But here's the good news: Authenticated requests get a higher rate limit. Check out the documentation for more details.) [rate reset in 11m19s]\n","time":"2020-03-11T13:13:05Z"}
Mar 11 13:13:05 192.168.80.1 trivy-adapter[21287]: {"error":"running trivy wrapper: running trivy: exit status 1: ","level":"error","msg":"Scan failed","time":"2020-03-11T13:13:05Z"}
Mar 11 13:13:06 192.168.80.1 trivy-adapter[21287]: {"exit_code":1,"image_ref":"core:8080/library/hello-world@sha256:ba27d460cd1f22a1a4331bdf74f4fccbc025552357e8a3249c40ae216275de96","level":"error","msg":"Running trivy failed","std_err":"","std_out":"2020-03-11T13:13:05.602Z\t\u001b[34mINFO\u001b[0m\tNeed to update DB\n2020-03-11T13:13:05.607Z\t\u001b[34mINFO\u001b[0m\tDownloading DB...\n2020-03-11T13:13:05.999Z\t\u001b[31mFATAL\u001b[0m\tfailed to download vulnerability DB: failed to download vulnerability DB: failed to list releases: GET https://api.github.com/repos/aquasecurity/trivy-db/releases: 403 API rate limit exceeded for 66.170.99.2. (But here's the good news: Authenticated requests get a higher rate limit. Check out the documentation for more details.) [rate reset in 11m18s]\n","time":"2020-03-11T13:13:06Z"}
Mar 11 13:13:06 192.168.80.1 trivy-adapter[21287]: {"error":"running trivy wrapper: running trivy: exit status 1: ","level":"error","msg":"Scan failed","time":"2020-03-11T13:13:06Z"}
Mar 11 13:13:06 192.168.80.1 trivy-adapter[21287]: {"error":"running trivy wrapper: running trivy: exit status 1: ","level":"error","msg":"Scan job failed","scan_job_id":"1df1fbd5dc8d126e41c25bc6","time":"2020-03-11T13:13:06Z"}
Mar 11 13:13:09 192.168.80.1 trivy-adapter[21287]: {"error":"running trivy wrapper: running trivy: exit status 1: ","level":"error","msg":"Scan job failed","scan_job_id":"468f7663070f5cf69b7c871a","time":"2020-03-11T13:13:09Z"}
Mar 11 13:13:12 192.168.80.1 trivy-adapter[21287]: {"exit_code":1,"image_ref":"core:8080/library/redis@sha256:ee1d00c5250b5a886b09be2d5f9506add35dfb557f1ef37a7e4b8f0138f32956","level":"error","msg":"Running trivy failed","std_err":"","std_out":"2020-03-11T13:13:12.143Z\t\u001b[34mINFO\u001b[0m\tNeed to update DB\n2020-03-11T13:13:12.143Z\t\u001b[34mINFO\u001b[0m\tDownloading DB...\n2020-03-11T13:13:12.534Z\t\u001b[31mFATAL\u001b[0m\tfailed to download vulnerability DB: failed to download vulnerability DB: failed to list releases: GET https://api.github.com/repos/aquasecurity/trivy-db/releases: 403 API rate limit exceeded for 66.170.99.2. (But here's the good news: Authenticated requests get a higher rate limit. Check out the documentation for more details.) [rate reset in 11m12s]\n","time":"2020-03-11T13:13:12Z"}
Mar 11 13:13:12 192.168.80.1 trivy-adapter[21287]: {"error":"running trivy wrapper: running trivy: exit status 1: ","level":"error","msg":"Scan failed","time":"2020-03-11T13:13:12Z"}
Mar 11 13:13:13 192.168.80.1 trivy-adapter[21287]: {"error":"running trivy wrapper: running trivy: exit status 1: ","level":"error","msg":"Scan job failed","scan_job_id":"b33fb182e3850d07cc56d9d6","time":"2020-03-11T13:13:13Z"}
Mar 11 13:13:24 192.168.80.1 trivy-adapter[21287]: {"exit_code":1,"image_ref":"core:8080/library/redis@sha256:158b81e0bd9bd3ec6794542f20d417aee58391f1dc719756cea55f74f51bc055","level":"error","msg":"Running trivy failed","std_err":"","std_out":"2020-03-11T13:13:23.681Z\t\u001b[34mINFO\u001b[0m\tNeed to update DB\n2020-03-11T13:13:23.681Z\t\u001b[34mINFO\u001b[0m\tDownloading DB...\n2020-03-11T13:13:24.092Z\t\u001b[31mFATAL\u001b[0m\tfailed to download vulnerability DB: failed to download vulnerability DB: failed to list releases: GET https://api.github.com/repos/aquasecurity/trivy-db/releases: 403 API rate limit exceeded for 66.170.99.2. (But here's the good news: Authenticated requests get a higher rate limit. Check out the documentation for more details.) [rate reset in 11m01s]\n","time":"2020-03-11T13:13:24Z"}
Mar 11 13:13:24 192.168.80.1 trivy-adapter[21287]: {"error":"running trivy wrapper: running trivy: exit status 1: ","level":"error","msg":"Scan failed","time":"2020-03-11T13:13:24Z"}
Mar 11 13:13:24 192.168.80.1 trivy-adapter[21287]: {"error":"running trivy wrapper: running trivy: exit status 1: ","level":"error","msg":"Scan job failed","scan_job_id":"696b9cab89ff2a46c3a3228a","time":"2020-03-11T13:13:24Z"}
Mar 11 13:13:40 192.168.80.1 trivy-adapter[21287]: {"exit_code":1,"image_ref":"core:8080/library/redis@sha256:16ece50399ba1eb8ed67e03687423434e5fa75c52a448f94519664e0a9e7829c","level":"error","msg":"Running trivy failed","std_err":"","std_out":"2020-03-11T13:13:40.240Z\t\u001b[34mINFO\u001b[0m\tNeed to update DB\n2020-03-11T13:13:40.241Z\t\u001b[34mINFO\u001b[0m\tDownloading DB...\n2020-03-11T13:13:40.637Z\t\u001b[31mFATAL\u001b[0m\tfailed to download vulnerability DB: failed to download vulnerability DB: failed to list releases: GET https://api.github.com/repos/aquasecurity/trivy-db/releases: 403 API rate limit exceeded for 66.170.99.2. (But here's the good news: Authenticated requests get a higher rate limit. Check out the documentation for more details.) [rate reset in 10m44s]\n","time":"2020-03-11T13:13:40Z"}
Mar 11 13:13:40 192.168.80.1 trivy-adapter[21287]: {"error":"running trivy wrapper: running trivy: exit status 1: ","level":"error","msg":"Scan failed","time":"2020-03-11T13:13:40Z"}
Mar 11 13:13:42 192.168.80.1 trivy-adapter[21287]: {"exit_code":1,"image_ref":"core:8080/library/redis@sha256:16ece50399ba1eb8ed67e03687423434e5fa75c52a448f94519664e0a9e7829c","level":"error","msg":"Running trivy failed","std_err":"","std_out":"2020-03-11T13:13:41.789Z\t\u001b[34mINFO\u001b[0m\tNeed to update DB\n2020-03-11T13:13:41.789Z\t\u001b[34mINFO\u001b[0m\tDownloading DB...\n2020-03-11T13:13:42.180Z\t\u001b[31mFATAL\u001b[0m\tfailed to download vulnerability DB: failed to download vulnerability DB: failed to list releases: GET https://api.github.com/repos/aquasecurity/trivy-db/releases: 403 API rate limit exceeded for 66.170.99.2. (But here's the good news: Authenticated requests get a higher rate limit. Check out the documentation for more details.) [rate reset in 10m43s]\n","time":"2020-03-11T13:13:42Z"}
Mar 11 13:13:42 192.168.80.1 trivy-adapter[21287]: {"error":"running trivy wrapper: running trivy: exit status 1: ","level":"error","msg":"Scan failed","time":"2020-03-11T13:13:42Z"}
Mar 11 13:13:42 192.168.80.1 trivy-adapter[21287]: {"error":"running trivy wrapper: running trivy: exit status 1: ","level":"error","msg":"Scan job failed","scan_job_id":"989774e8da5bc572dd1486d0","time":"2020-03-11T13:13:42Z"}
Mar 11 13:13:43 192.168.80.1 trivy-adapter[21287]: {"error":"running trivy wrapper: running trivy: exit status 1: ","level":"error","msg":"Scan job failed","scan_job_id":"85c0d0d807fa5db268ed6a54","time":"2020-03-11T13:13:43Z"}
Mar 11 13:13:48 192.168.80.1 trivy-adapter[21287]: {"exit_code":1,"image_ref":"core:8080/library/hello-world@sha256:ba27d460cd1f22a1a4331bdf74f4fccbc025552357e8a3249c40ae216275de96","level":"error","msg":"Running trivy failed","std_err":"","std_out":"2020-03-11T13:13:48.327Z\t\u001b[34mINFO\u001b[0m\tNeed to update DB\n2020-03-11T13:13:48.329Z\t\u001b[34mINFO\u001b[0m\tDownloading DB...\n2020-03-11T13:13:48.728Z\t\u001b[31mFATAL\u001b[0m\tfailed to download vulnerability DB: failed to download vulnerability DB: failed to list releases: GET https://api.github.com/repos/aquasecurity/trivy-db/releases: 403 API rate limit exceeded for 66.170.99.2. (But here's the good news: Authenticated requests get a higher rate limit. Check out the documentation for more details.) [rate reset in 10m36s]\n","time":"2020-03-11T13:13:48Z"}
```

**Comment by reasonerjt** [comment]
@danielpacak 
I know this is rare case but I think we need a release-note to explain this situation and provide a reasonable work around.

We used to have issue that a company was banned because many teams are using Harbor and share the gateway to download.
So this may happen more often than you expect.

**Comment by danielpacak** [comment]
I've opened a PR to allows admis to set GitHub token is rate limit is a concern; I'll merge and release soon: https://github.com/aquasecurity/harbor-scanner-trivy/pull/80

If it does not solve the problem, we could think about other means of downloading the DB file from GitHub releases page: https://github.com/aquasecurity/trivy-db/releases, and then use the `--skip-download` flag in Trivy.

At the end of the day a single instance of the Trivy scanner adapter is not supposed to download the DB file from GitHub that often.

/cc @knqyf263 @lizrice

**Comment by heww** [comment]
Closed by #11047
